gkeys/gkeys/actions.py

```diff
@@ -351,8 +376,32 @@ def checkkey(self, args):
                         % (gkey.name, gkey.nick, key))
                 if not results[gkey.name].sign:
                     failed['sign'].append(_unicode("%s <%s>: %s ")
-                        % (gkey.name, gkey.nick, key))
-        if failed['expired']:
+                        % (gkey.name, gkey.nick, key))'''
+        print (keys_expiring)
+        mail = smtplib.SMTP('smtp.gmail.com', 587)
+        mail.ehlo()
+        mail.starttls()
+        mail.login('aperivol@gmail.com', 'PASSWORD')
+        c = 0
+        for email in keys_expiring:
+            if c%5 == 0:
+                print(email)
+                expire_key = keys_expiring[email]
+                if expire_key[1] < 0:
+                    expire = "has expired %d days ago" %(-1*expire_key[1])
+                else:
+                    expire = "is expiring in %d days" %expire_key[1]
+                subject = "Expiring Key"
+                email_to = "justthisthing@gmail.com"
+                email_from = "aperivol@gmail.com"
+                message = "Your gentoo key %s %s" %(str(expire_key[0]), expire)
+                msg = MIMEText(message, 'plain')
+                msg['Subject'] = subject
+                msg['From'] = email_from
+                mail.sendmail('aperivol@gmail.com', 'justthisthing@gmail.com', msg.as_string())
+            c += 1
+        mail.quit()
```

Comment:
@dol-sen this is an outdated diff, I was testing it there at first but then reworked the whole logic behind it.